### PR TITLE
Enable more linting

### PR DESCRIPTION
**Pull request by zanderso**
## Description

Enables linting under `runtime/` with some adjustments under `shell/`.

## Breaking Change

Did any tests fail when you ran them? Please read [handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

### CODE REVIEW

File: runtime/dart_isolate.cc [hunk 1]
```diff
@@ -689,7 +692,8 @@ Dart_Isolate DartIsolate::DartIsolateGroupCreateCallback(
               parent_group_data.GetIsolateShutdownCallback())));
 
   TaskRunners null_task_runners(advisory_script_uri,
-                                /* platform= */ nullptr, /* raster= */ nullptr,
+                                /* platform= */ nullptr,
+                                /* raster= */ nullptr,
```
Comment (gaaclarke) [inline runtime/dart_isolate.cc:696]: The C++ style is to not have the spaces between the comments: https://google.github.io/styleguide/cppguide.html#Function_Argument_Comments

I'm not sure if the linter is smart enough to handle the spaces.

File: runtime/dart_isolate_unittests.cc [hunk 2]
```diff
@@ -153,11 +164,10 @@ TEST_F(DartIsolateTest, CanRunDartCodeCodeSynchronously) {
 
 TEST_F(DartIsolateTest, CanRegisterNativeCallback) {
   ASSERT_FALSE(DartVMRef::IsInstanceRunning());
-  fml::AutoResetWaitableEvent latch;
   AddNativeCallback("NotifyNative",
-                    CREATE_NATIVE_ENTRY(([&latch](Dart_NativeArguments args) {
+                    CREATE_NATIVE_ENTRY(([&](Dart_NativeArguments args) {
```
Comment (gaaclarke) [inline runtime/dart_isolate_unittests.cc:168]: FYI you can probably just capture `[this]`, not a huge deal for tests.

File: runtime/dart_isolate.cc (hunk 1 above)
Comment (zanderso) [reply runtime/dart_isolate.cc:696]: I hope it is, because the formatter is adding them =)

File: runtime/dart_isolate_unittests.cc (hunk 2 above)
Comment (zanderso) [reply runtime/dart_isolate_unittests.cc]: Done.

**Pull request merged**